tests/scripts/all.sh

```diff
@@ -4162,6 +4162,26 @@ component_test_aes_fewer_tables_and_rom_tables () {
     make test
 }
 
+component_test_aes_encrypt_only () {
+    msg "build: default config + PSA_CRYPTO_CONFIG + implicitly enable AES_ENCRYPT_ONLY"
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py unset MBEDTLS_CIPHER_MODE_CBC
+    scripts/config.py unset MBEDTLS_CIPHER_MODE_XTS
+    scripts/config.py unset MBEDTLS_NIST_KW_C
+
+    echo '#undef PSA_WANT_ALG_CBC_NO_PADDING' >> psa_aes_encrypt_only.h
+    echo '#undef PSA_WANT_ALG_CBC_PKCS7' >> psa_aes_encrypt_only.h
+    echo '#undef PSA_WANT_ALG_ECB_NO_PADDING' >> psa_aes_encrypt_only.h
+
+    make CC=gcc CFLAGS="-Werror -Wall -Wextra -I '$PWD' \
+        -DMBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE='\"psa_aes_encrypt_only.h\"'"
```

Comment:
This only test on the AESNI assembly implementation.
To improve the test coverage, we need to test on all the implementations(AESNI, AESCE, VIA Padlock).